### PR TITLE
Initial standalone Simulator GUI

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description

Adds a simple GUI for the standalone simulator that shows the state of the "world" with the data available in an SSL_WrapperPacket. See slack for an example of what this looks like.

This PR adds an "example main" binary for the simulator that connects the visualization to an instance of a `ThreadedSimulator` to show them working together. I don't think the `gui` folder is the best place for the `Simulator` binary (really should be in `simulation` I think), but am open to ideas.

I'm also open to feedback on the names I have currently. We already have a class named `Simulator`, so I had to differentiate for the GUI. I'm not sure how I feel about the names `SimulatorGUI` and `SimulatorGUIWrapper`, so please feel free to give naming idea.
* Perhaps `StandaloneSimulator`?
* Perhaps `NetworkedSimulator`?

### Testing Done

Can't really unit tests GUI stuff, but verified manually it works as expected.

### Length Justification

it's _barely_ over

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/gui/simulator/widgets/simulator_gui.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    // the Qt system takes ownership of the pointer and is responsible for
+    // de-allocating it so we don't have to
+    MainWidget* main_widget;
+    QTimer* update_timer;
+
+    std::shared_ptr<ThreadSafeBuffer<SSL_WrapperPacket>> ssl_wrapper_packet_buffer;
+    std::shared_ptr<ThreadSafeBuffer<Rectangle>> view_area_buffer;
+
+    static constexpr double DRAW_UPDATE_INTERVAL_SECONDS = 1.0 / 60.0;
+};
```
Comment (garethellis0) [inline src/software/gui/simulator/widgets/simulator_gui.h]: missing newline at eof

**Review by garethellis0 — CHANGES_REQUESTED** [review]
one super easy comment

File: src/software/gui/simulator/widgets/simulator_gui.h (hunk 1 above)
Comment (MathewMacDougall) [reply src/software/gui/simulator/widgets/simulator_gui.h]: added

**Review by garethellis0 — DISMISSED** [review]

**Comment by MathewMacDougall** [comment]
Renamed `SimulatorGUI` to `StandaloneSimulatorGUI`, and moved the `standalone_simulator` binary to the `simulation` folder.

**Review by garethellis0 — APPROVED** [review]

File: src/software/gui/standalone_simulator/standalone_simulator_gui_wrapper.cpp [hunk 2]
```diff
@@ -0,0 +1,87 @@
+#include "software/gui/standalone_simulator/standalone_simulator_gui_wrapper.h"
+
+#include <QtCore/QTimer>
+#include <QtWidgets/QApplication>
+
+#include "software/gui/standalone_simulator/widgets/standalone_simulator_gui.h"
+#include "software/proto/message_translation/ssl_geometry.h"
+
+StandaloneSimulatorGUIWrapper::StandaloneSimulatorGUIWrapper(int argc, char** argv)
```
Comment (jonathanlew) [inline src/software/gui/standalone_simulator/standalone_simulator_gui_wrapper.cpp:9]: Could we (not in this PR) template GuiWrapper?

**Review by jonathanlew — APPROVED** [review]
:tennis:

File: src/software/gui/standalone_simulator/standalone_simulator_gui_wrapper.cpp (hunk 2 above)
Comment (MathewMacDougall) [reply src/software/gui/standalone_simulator/standalone_simulator_gui_wrapper.cpp:9]: Sure, what are you thinking the template arg(s) would be?

**Review by kelvinkoon — APPROVED** [review]

File: src/software/gui/standalone_simulator/standalone_simulator_gui_wrapper.cpp (hunk 2 above)
Comment (jonathanlew) [reply src/software/gui/standalone_simulator/standalone_simulator_gui_wrapper.cpp:9]: hm, not much to template, but a lot of repeated code that could go into a super class

Comment (jonathanlew) [reply src/software/gui/standalone_simulator/standalone_simulator_gui_wrapper.cpp:9]: hm, not really templateable but we should be able to pull out the duplicated code

Comment (MathewMacDougall) [reply src/software/gui/standalone_simulator/standalone_simulator_gui_wrapper.cpp:9]: Yea will look at cleaning up / simplifying all the GUI stuff soon once simple versions of everything are running

**Pull request merged**